File: pkg/models/installed_app.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// AppID constants for known applications.
const (
	AppIDMCPServer     = "mcp-server"
	AppIDOntologyForge = "ontology-forge"
	AppIDAIDataLiaison = "ai-data-liaison"
	AppIDAIAgents      = "ai-agents"
	AppIDFileLoader    = "file-loader"
	AppIDMCPTunnel     = "mcp-tunnel"
)

// KnownAppIDs is the set of valid app identifiers.
var KnownAppIDs = map[string]bool{
	AppIDMCPServer:     true,
	AppIDOntologyForge: true,
	AppIDAIDataLiaison: true,
	AppIDAIAgents:      true,
	AppIDFileLoader:    true,
	AppIDMCPTunnel:     true,
}

// InstalledApp represents an application installed for a project.
type InstalledApp struct {
	ID          uuid.UUID      `json:"id"`
	ProjectID   uuid.UUID      `json:"project_id"`
	AppID       string         `json:"app_id"`
	InstalledAt time.Time      `json:"installed_at"`
	InstalledBy string         `json:"installed_by,omitempty"`
	ActivatedAt *time.Time     `json:"activated_at,omitempty"`
	Settings    map[string]any `json:"settings"`
}

File: ui/src/types/installedApp.ts
/**
 * Installed App Types
 * Types for tracking installed applications per project
 */

/**
 * Known application IDs
 */
export const APP_ID_MCP_SERVER = 'mcp-server';
export const APP_ID_ONTOLOGY_FORGE = 'ontology-forge';
export const APP_ID_AI_DATA_LIAISON = 'ai-data-liaison';
export const APP_ID_AI_AGENTS = 'ai-agents';
export const APP_ID_FILE_LOADER = 'file-loader';
export const APP_ID_MCP_TUNNEL = 'mcp-tunnel';

/**
 * Installed application record
 */
export interface InstalledApp {
  id: string;
  project_id: string;
  app_id: string;
  installed_at: string;
  installed_by?: string;
  activated_at?: string;
  settings: Record<string, unknown>;
}

/**
 * Response from list installed apps endpoint
 */
export interface InstalledAppsResponse {
  apps: InstalledApp[];
}
